src/main/kotlin/org/ionproject/integration/extractor/interface/PdfExtractor.kt

```diff
@@ -0,0 +1,13 @@
+package org.ionproject.integration.extractor.`interface`
+
+import org.ionproject.integration.utils.Try
+
+interface PdfExtractor {
+    /**
+     * Extract data from pdf file locate at [pdfPath]
+     * @return Try<MutableList<String>>
+     *     Success - String list contains extracted data
+     *     Failure - PdfExtractorException
+     */
+    suspend fun extract(pdfPath: String): Try<MutableList<String>>
```

Comment:
Why a mutable list?

Reply:
What is this  `pdfPath`? Can't it be a more strongly-typed _thing_ (e.g `URI` or `File`)?

Reply:
Data obtained via `iText` comes in string format and it's easier to process the data page by page.
`pdfPath` is the path to the pdf file. It has to be a path because `Tabula` is executed as CLI.

Reply:
From the following sentence present at [https://github.com/tabulapdf/tabula-java/], I got the idea that it is possible to use Tabula as a library
```
You can also integrate tabula-java with any JVM language. For Java examples, see the tests folder.
```

Reply:
From what I was able to gather we should use `CommandLineApp` and that receives a file path. The alternative would be to use [PDFBox](https://pdfbox.apache.org/) that [tabula uses](https://github.com/tabulapdf/tabula-java/blob/master/pom.xml#L238), but that would negate the benefits of using `tabula`...